pages/api/group/index.ts

```diff
@@ -0,0 +1,113 @@
+import mongoose from 'mongoose';
+import type { NextApiRequest, NextApiResponse } from 'next';
+import {dbConnect} from '@/lib/mongoClient';
+import { responseFormatter } from '@/lib/responseLib';
+import Group from '@/models/Group';
+import  UserGroupRelation  from '@/models/UserGroupRelation';
+
+function getRandomID(name:string):string {
+    let ID = name.split(" ").join("-") + "-";
+    for(let i=0; i<4; i++){
+        ID += Math.floor(Math.random() * 10).toString();
+    }
+    return ID
+  }
+
+export default async function groupHandler(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  // Connect with MongoDB database
+  try {
+    await dbConnect();
+  } catch (e) {
+    console.log('fail to connect with database', e);
+    return (
+      res
+        .status(500)
+        .json(responseFormatter(false, null, 'Internal Server Error'))
+    );
+  }
+
+  switch (req.method) {
+    // create a new group
+    case 'POST':
+      try {
+        const group = req.body;
```

Comment:
It will be better if we could use a schema validation library to validate all `write` HTTP request body, but again it is not necessary